database/src/main/java/com/firebase/ui/database/FirebaseArray.java

```diff
@@ -25,7 +25,7 @@
  * This class implements an array-like collection on top of a Firebase location.
  */
 class FirebaseArray implements ChildEventListener {
-    public interface OnChangedListener {
```

Comment:
This one should definitely be changed, I don't know why it's not at least protected.